static/js/public/store-categories.js

```diff
@@ -0,0 +1,43 @@
+import Carousel from "./carousel";
+import 'whatwg-fetch';
+
+/**
+ *
+ * @param {HTMLElement} holder
+ * @returns {Promise<string>}
+ */
+function getCategory(holder) {
+  const category = holder.dataset.category;
+
+  // Write the html and init the carousel
+  const writeCategory = function(response) {
+    holder.innerHTML = response;
+    new Carousel(holder.querySelector('.p-carousel'));
+  };
+
+  const url = `/store/categories/${category}`;
+
+  // Use fetch ¯\_(ツ)_/¯
+  return fetch(url).then(response => {
+    if (!response.ok) {
+      throw `Response not ok for ${category} category \`${url}\``;
```

Comment:
I don't know what is best in case an api call doesn't work, diplay a message or not?

**Example all api fail**
![image](https://user-images.githubusercontent.com/2707508/42211913-964d89e6-7eac-11e8-9779-b4722b085319.png)

Reply:
I specifically made it like this so that it would just be an 'ok' list if the API fails - interested in a UX perspective though

Reply:
Probably makes sense to have this list still and work out a little bit more on the edge cases as we go along, as this is a first iteration pass of the store categories feature